src/library/libraryview.cpp

```diff
@@ -642,11 +643,19 @@ void LibraryView::Organise() {
 }
 
 void LibraryView::Delete() {
-  if (QMessageBox::warning(this, tr("Delete files"),
-                           tr("These files will be permanently deleted from "
-                              "disk, are you sure you want to continue?"),
-                           QMessageBox::Yes,
-                           QMessageBox::Cancel) != QMessageBox::Yes)
+
+  const SongList selected_songs = GetSelectedSongs();
+
+  QString message = tr("The following files will be permanently deleted from "
+                              "disk:") +
+                      "<ul>";
+  for (const Song& song : selected_songs) {
+    message += ("<li>" + song.url().toString() + "</li>");
```

Comment:
I think you should use one of Qt's XML writers to construct this html to get the escaping right. You can also omit the html from the translation strings if you do that.